sdks/java/extensions/gcp-core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -436,8 +436,16 @@ public SeekableByteChannel open(GcsPath path)
    * @param type the type of object, eg "text/plain".
    * @return a Callable object that encloses the operation.
    */
-  public WritableByteChannel create(GcsPath path,
-      String type) throws IOException {
+  public WritableByteChannel create(GcsPath path, String type) throws IOException {
+    return create(path, type, uploadBufferSizeBytes);
+  }
+
+  /**
+   * Same as {@link GcsUtil#create(GcsPath, String)} but allows overriding
+   * {code uploadBufferSizeBytes}.
+   */
+  public WritableByteChannel create(GcsPath path, String type, Integer uploadBufferSizeBytes)
```

Comment:
One option is to move `GcsCreateOptions` to `sdks/java/extensions/gcp-core` (where `GcsUtil` resides), and that way there won't be circular dependencies between `sdks/java/gcp` and `sdks/java/extensions/gcp-core`. But I am still hesitant to have `GcsUtil` depend on any `sdk.io.fs` just for create options, maybe worth taking a look later into unifying options for all the operations?